Assignment4/JorgeKGIT-a180241/task08.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+"""Task08.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1b9lk0UIy_jejWFkW0yzGBdL2r9FO6_nE
+
+**Task 08: Completing missing data**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+from rdflib import Graph, Namespace, Literal, URIRef
+g1 = Graph()
+g2 = Graph()
+g1.parse(github_storage+"/rdf/data01.rdf", format="xml")
+g2.parse(github_storage+"/rdf/data02.rdf", format="xml")
+
+from rdflib.namespace import RDF, RDFS
+g1.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g1.namespace_manager.bind('data', Namespace("http://data.org#"), override=False)
+data = Namespace("http://data.org#")
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+print(vcard.Given)
+
+"""Tarea: lista todos los elementos de la clase Person en el primer grafo (data01.rdf) y completa los campos (given name, family name y email) que puedan faltar con los datos del segundo grafo (data02.rdf). Puedes usar consultas SPARQL o iterar el grafo, o ambas cosas."""
+
+for s,p,o in g1.triples((None,RDF.type,data.Person)):
+  for s1,p1,o1 in g1.triples((s,None,None)):
+    print(s1,p1,o1)
+
+for s,p,o in g2.triples((None,RDF.type,data.Person)):
+  for s1,p1,o1 in g2.triples((s,None,None)):
+
+    if p1 ==vcard.Given or p1==vcard.EMAIL or p1==vcard.Family:
+      print(s1,p1,o1)
+      g1.add((s1, p1, o1))
```

Comment:
The exercise is asking you to complete the name, family name and email from g1 with g2. But you are adding ALL the elements from g2 in g1 that have a name, email or family name.  We want to complete the info from g1 individuals without adding g2 individuals